nifi-mock/src/main/java/org/apache/nifi/util/MockControllerServiceInitializationContext.java

```diff
@@ -70,6 +71,11 @@ public StateManager getStateManager() {
         return stateManager;
     }
 
+    @Override
+    public NodeTypeProvider getNodeTypeProvider() {
+        return null;
```

Comment:
I think we probably want a way to utilize the NodeTypeProvider from the mock framework. Looking at MockProcessorInitializationContext, it returns a MockProcessContext which implements NodeTypeProvider and also has methods like setClustered and setPrimary to simulate different scenarios. 

We probably wouldn't use MockProcessContext here, but maybe we can make MockControllerServiceInitializationContext implement NodeType provider + have the setter methods, and then getNodeTypeProvider() returns this ?

Reply:
@bbende Thanks for catching that. I would have probably been back here submitting a PR for that in a week or two the first time I tried to actually write a unit test that uses this :) I've updated the code.